da/avail/datasubmit/submitdata.go

```diff
@@ -0,0 +1,89 @@
+package datasubmit
+
+import (
+	"errors"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+// submit data submits the extrinsic through substrate api
+func SubmitData(apiURL string, seed string, appID int, data []byte) error {
+	api, err := gsrpc.NewSubstrateAPI(apiURL)
+	if err != nil {
+		return err
+	}
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		return err
+	}
+
+	//if app id is greater than 0 then it must be created before submitting data
+	err = errors.New("AppID cant be 0")
+	if appID == 0 {
+		return err
+	}
```

Comment:
```suggestion
	if appID == 0 {
		return errors.New("AppID cannot be 0")
	}
```

And check this on the top of this function